### PR TITLE
MM-30371: Cypress/E2E: Automate backlogs - Keyboard Shortcuts part 2 (10 test cases)

### PR DESCRIPTION
**Pull request by catalintomai**
#### Summary
Cypress/E2E: Automate backlogs - Keyboard Shortcuts part 2 (10 test cases)

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-30371

### CODE REVIEW

**Comment by catalintomai** [comment]
@saturninoabril 
MM-T1258 and MM-T1259 are missing from the PR.

MM-T1258 requires access to the native file uploader. According to https://github.com/cypress-io/cypress/issues/170 there is no current Cypress support - there is a plugin that fills the gap https://github.com/abramenal/cypress-file-upload , but not clear how it should be used/invoked when the shortcut is typed. 
Any ideas or should we treat this test case as a N/A wrt to Cypress?

MM-T1239 - for "CMD+["/"CMD+]" I tried many variations of this (with/without delay, wait time, etc.)
`cy.get('#post_textbox').type('{cmd}', {release: false}).type(']').type('{cmd}', {release: true}).wait(TIMEOUTS.HALF_SEC);`
 none worked so far (tried it on the Mac only). 
Any ideas of what else I can try or should we treat this test case as N/A wrt to Cypress?

cc: @DHaussermann

**Comment by saturninoabril** [comment]
Thanks for trying @catalintomai! For file upload, looks like there's no other workaround. For MM-T1239, I'll play around once I get a chance to review this PR, probably later today.

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- Test actions look correct
- All tests run Green
- Set all corresponding TM4J ticket Cypress status to `Update`

LGTM!

File: e2e/cypress/integration/keyboard_shortcuts/keyboard_shortcuts_two_spec.js [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+
+            // # Press CTRL/CMD+SHIFT+UP
+            if (cy.isMac()) {
+                cy.get('body').type('{cmd}{option}', {release: false}).type('{uparrow}').type('{cmd}{option}', {release: true});
+            } else {
+                cy.get('body').type('{ctrl}{shift}', {release: false}).type('{uparrow}').type('{ctrl}{shift}', {release: true});
+            }
+        }
+
+        // Step #2 - not clear what are the "new shortcuts" - verifying that 'Keyboard Shortcuts' show
```
Comment (saturninoabril) [inline e2e/cypress/integration/keyboard_shortcuts/keyboard_shortcuts_two_spec.js:259]: I believe it's to verify if shortcuts for "Previous Team" "Next Team" and "Switch to a specific team" are displayed in Keyboard shortcuts dialog. I've updated the test case.
Screen Shot 2020-12-02 at 11 51 25 PM

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks Catalin! Please see comments.

File: e2e/cypress/integration/keyboard_shortcuts/keyboard_shortcuts_two_spec.js [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+        }
+
+        for (let index = 0; index < count; index++) {
+            // # Verify that we've switched to the correct team
+            cy.get('#headerTeamName').should('contain', teamDisplayNames[count - index - 1]);
+
+            // # Verify that we've switched to the correct channel
+            cy.get('#channelHeaderTitle').should('be.visible').should('contain', channelDisplayNames[count - index - 1]);
+
+            // # Press CTRL/CMD+SHIFT+UP
```
[the remark this answers is not in the document]
Comment (catalintomai) [reply e2e/cypress/integration/keyboard_shortcuts/keyboard_shortcuts_two_spec.js:249]: My bad, should be present now.

**Review by saturninoabril — APPROVED** [review]
@catalintomai Thanks for updating! LGTM.

**Pull request merged**